File: app/src/main/java/com/typ/muslim/enums/PrayStatus.java
/*
 * This app is developed by AHMED SLEEM
 *
 * Copyright (c) 2021.  TYP INC. All Rights Reserved
 */

package com.typ.muslim.enums;

public enum PrayStatus {

	UP_COMING,
	ON_TIME,
	DELAYED,
	FORGOT;

	public static PrayStatus valueOf(int ordinal) {
		return ordinal == 0 ? UP_COMING : ordinal == 1 ? ON_TIME : ordinal == 2 ? DELAYED : FORGOT;
	}

}

File: app/src/main/java/com/typ/muslim/features/prays/enums/PrayStatus.kt
/*
 * This app is developed by AHMED SLEEM
 *
 * Copyright (c) 2021.  TYP INC. All Rights Reserved
 */
package com.typ.muslim.features.prays.enums

enum class PrayStatus {
    UP_COMING,
    ON_TIME,
    DELAYED,
    FORGOT;

    companion object {
        @JvmStatic
        fun valueOf(ordinal: Int): PrayStatus {
            return when (ordinal) {
                0 -> UP_COMING
                1 -> ON_TIME
                2 -> DELAYED
                else -> FORGOT
            }
        }
    }
}
